core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -72,6 +72,10 @@ oppia.factory('AssetsBackendApiService', [
     };
 
     var _getAudioDownloadUrl = function(explorationId, filename) {
+      if (GLOBALS.DEV_MODE) {
+        return '/assets/test/' + filename;
```

Comment:
OK, I thought about this. It's not straightforward, unfortunately. Here's the reasoning:

constants.js needs to be a static dict (since the backend interprets it as JSON), so it can't have any logic in it. That's a shame, because that's the file for constants shared between frontend and backend. But we can't have an if/else condition there.

The other place we store constants is in feconf, and we can do "if DEV_MODE ... else ..." there. We can then ship it to the frontend as a GLOBALS property -- see how ALLOWED_GADGETS is sent in editor.py, for example. The logic here should then be:

`return UrlInterpolationService.interpolateUrl(GLOBALS.AUDIO_URL_PREFIX, {...});`

This would be better than doing a one-off thing here, because that way there'll be a single audio url prefix used everywhere. This becomes important when we work on the editor page because audio files need to be uploaded to that same location as well, so this prefix needs to be somehow global.